PubNub/Network/Parsers/PNSubscribeParser.m

```diff
@@ -152,35 +160,16 @@ + (NSDictionary *)parsedServiceResponse:(id)response withData:(NSDictionary *)ad
     // through 'nil' initialized local variable.
     NSDictionary *processedResponse = nil;
     
-    // Array will arrive in case of subscription event
-    if ([response isKindOfClass:[NSArray class]]) {
-        
-        NSArray *feedEvents = response[kPNEventsListElementIndex];
-        NSNumber *timeToken = @([response[kPNEventTimeTokenElement] longLongValue]);
-        NSArray *channels = nil;
-        NSArray *groups = nil;
-        if ([(NSArray *)response count] > kPNEventChannelsElementIndex) {
-            
-            channels = [PNChannel namesFromRequest:response[kPNEventChannelsElementIndex]];
-        }
-        if ([(NSArray *)response count] > kPNEventChannelsDetailsElementIndex) {
-            
-            groups = [PNChannel namesFromRequest:response[kPNEventChannelsDetailsElementIndex]];
-        }
-        
-        // Checking whether at least one event arrived or not.
+    // response is subscription
+    if ([response isKindOfClass:[NSDictionary class]]) {
+        NSNumber *timeToken = @([response[kPNEventsTimeTokenElementKey][kPNEventsTimeTokenElementKey] longLongValue]);
+        NSArray *feedEvents = response[kPNEventsMessageElementKey];
         if ([feedEvents count]) {
-            
             NSMutableArray *events = [[NSMutableArray alloc] initWithCapacity:[feedEvents count]];
             for (NSUInteger eventIdx = 0; eventIdx < [feedEvents count]; eventIdx++) {
                 
                 // Fetching remote data object name on which event fired.
-                NSString *objectOrGroupName = (eventIdx < [channels count] ? channels[eventIdx] : channels[0]);
-                NSString *objectName = ([groups count] > eventIdx ? groups[eventIdx] : nil);
-                NSMutableDictionary *event = [self eventFromData:feedEvents[eventIdx]
-                                                      forChannel:(objectName?: objectOrGroupName)
-                                                           group:(objectName? objectOrGroupName: nil)
-                                        withAdditionalParserData:additionalData];
+                NSMutableDictionary *event = [self eventFromData:feedEvents[eventIdx] withAdditionalParserData:additionalData];
                 event[@"timetoken"] = timeToken;
```

Comment:
@jzucker2 My note from above still valid and we should provide here global time token only if parsed event wasn't able to get one from packet envelop.